src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -25,131 +18,72 @@ HOSTFXR_UTILITY::~HOSTFXR_UTILITY()
 //
 HRESULT
 HOSTFXR_UTILITY::GetStandaloneHostfxrParameters(
-    PCWSTR              pwzExeAbsolutePath, // includes .exe file extension.
-    PCWSTR				pcwzApplicationPhysicalPath,
-    PCWSTR              pcwzArguments,
-    HANDLE              hEventLog,
-    _Inout_ STRU*		pStruHostFxrDllLocation,
-    _Out_ DWORD*		pdwArgCount,
-    _Out_ BSTR**		ppwzArgv
+        PCWSTR              pwzExeAbsolutePath, // includes .exe file extension.
+        PCWSTR				pcwzApplicationPhysicalPath,
+        PCWSTR              pcwzArguments,
+        HANDLE              hEventLog,
+        _Inout_ STRU*		pStruHostFxrDllLocation,
+        _Out_ DWORD*		pdwArgCount,
+        _Out_ BSTR**		ppwzArgv
 )
 {
-    HRESULT             hr = S_OK;
-    STRU                struDllPath;
-    STRU                struArguments;
-    STRU                struHostFxrPath;
-    STRU                struRuntimeConfigLocation;
-    DWORD               dwPosition;
-
-    // Obtain the app name from the processPath section.
-    if (FAILED(hr = struDllPath.Copy(pwzExeAbsolutePath)))
-    {
-        goto Finished;
-    }
+    HRESULT hr = S_OK;
 
-    dwPosition = struDllPath.LastIndexOf(L'.', 0);
-    if (dwPosition == -1)
-    {
-        hr = E_FAIL;
-        goto Finished;
-    }
+    const fs::path exePath(pwzExeAbsolutePath);
 
-    hr = UTILITY::ConvertPathToFullPath(L".\\hostfxr.dll", pcwzApplicationPhysicalPath, &struHostFxrPath);
-    if (FAILED(hr))
+    if (!exePath.has_extension())
     {
-        goto Finished;
+        return false;
     }
 
-    struDllPath.QueryStr()[dwPosition] = L'\0';
-    if (FAILED(hr = struDllPath.SyncWithBuffer()))
-    {
-        goto Finished;
-    }
+    const fs::path physicalPath(pcwzApplicationPhysicalPath);
+    const fs::path hostFxrLocation = physicalPath / L"hostfxr.dll";
```

Comment:
Though I know this is operated overloaded, I think it is fairly confusing. Use path.append()?

Reply:
Good point.